code/modules/reagents/chemistry/reagents/toxins.dm

```diff
@@ -1228,15 +1228,17 @@
 	color = "#20324D" //RGB: 32, 50, 77
 	metabolization_rate = 0.2
 	var/shock_timer = 0
+	var/normal_teslium = TRUE //Is this the base teslium, or the blob teslium. Used so it can be a subtype of teslium without inheriting the random shock from normal teslium as well as it's own
 	process_flags = ORGANIC | SYNTHETIC
 	taste_description = "electricity"
 
 /datum/reagent/teslium/on_mob_life(mob/living/M)
-	shock_timer++
-	if(shock_timer >= rand(5,30)) //Random shocks are wildly unpredictable
-		shock_timer = 0
-		M.electrocute_act(rand(5, 20), "Teslium in their body", 1, SHOCK_NOGLOVES) //Override because it's caused from INSIDE of you
-		playsound(M, "sparks", 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	if(normal_teslium)
```

Comment:
I feel like it would be better to have a var for minimum shock timer or something. Or a proc which handles the shocking.
Both implementations are the same except for the `chosen_timer` part

Reply:
I agree here. Please make it without an entire override with an if that cuts out half the proc, otherwise you may as well make it a new type instead of a subtype.

Reply:
Made it a proc